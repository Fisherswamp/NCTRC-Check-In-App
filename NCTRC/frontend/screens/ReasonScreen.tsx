import React from 'react';
import { SafeAreaView, StyleSheet, Text, View, Button, Image } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';
import { AuthStackParamList, AppScreens } from '../index';
type ReasonScreenNavigationProps = StackNavigationProp<AuthStackParamList, AppScreens.Reason>;
interface ReasonScreenProps {
    navigation: ReasonScreenNavigationProps;
}
const styles = StyleSheet.create({});
const ReasonScreen: React.FunctionComponent<ReasonScreenProps> = (props) => {
    const { navigation } = props;
    return (
        <SafeAreaView>
            <View>
<<<<<<< HEAD
                <Image source={require('./../assets/NCTRClogo.png')} style={{width: 400, height: 400}}></Image>
                <Text>What is the reason for your visit?</Text>
                <Button color='#884633' title="Employee" onPress={() => navigation.navigate(AppScreens.SignIn)} />
                <Button color='#884633' title="Client" onPress={() => navigation.navigate(AppScreens.SignIn)} />
                <Button color='#884633' title="Volunteer" onPress={() => navigation.navigate(AppScreens.SignIn)} />
                <Button color='#884633' title="Guest" onPress={() => navigation.navigate(AppScreens.SignIn)} />
                <Button color='#884633' title="Graston" onPress={() => navigation.navigate(AppScreens.SignIn)} />
=======
                <Image source={require('./../assets/NCTRClogo.png')} style={{ width: 400, height: 400 }}></Image>
                <Text>reason screen</Text>
                <Button color="#884633" title="Home" onPress={() => navigation.navigate(AppScreens.Home)} />
>>>>>>> 1509b0b4fe258ef4794c650b616003b9e42a13d5
            </View>
        </SafeAreaView>
    );
};
export default ReasonScreen;
