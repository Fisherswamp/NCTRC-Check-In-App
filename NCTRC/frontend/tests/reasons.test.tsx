
import React from "react";
import  ReasonScreen from '../screens/ReasonScreen';
import { shallow, configure } from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const props = {
    navigation: {
        addListener: jest.fn(), 
        canGoBack: jest.fn(), 
        dangerouslyGetParent: jest.fn(), 
        dangerouslyGetState: jest.fn(), 
        dispatch: jest.fn(), 
        goBack: jest.fn(), 
        isFocused: jest.fn(), 
        navigate: jest.fn(), 
        pop: jest.fn(), 
        popToTop: jest.fn(), 
        push: jest.fn(), 
        removeListener: jest.fn(), 
        replace: jest.fn(), 
        reset: jest.fn(), 
        setOptions: jest.fn(), 
        setParams: jest.fn(), 
    }, 
}

describe('Reasons renders', () => {
    it('Reasons render', () => {
      const tree = renderer.create(<ReasonScreen navigation={props.navigation} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});


describe('ReasonScreen elements', () => {
    it("ReasonScreen elements", () => {
       let wrapper =  shallow(<ReasonScreen navigation={props.navigation} />);
       expect(wrapper.find('Image').length).toBe(2); 
       expect(wrapper.find('Text').length).toBe(7); 
       expect(wrapper.find('View').length).toBe(1); 
    })
   
});

jest.useFakeTimers(); 