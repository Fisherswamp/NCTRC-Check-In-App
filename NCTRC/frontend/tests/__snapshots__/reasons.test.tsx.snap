// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reasons renders Reasons render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/backbutton.png",
        }
      }
      style={
        Object {
          "height": 75,
          "width": 75,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "marginTop": -40,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/NCTRClogo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
          "margin": 10,
        }
      }
    >
      What is the reason for your visit?
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 10,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Employee
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 10,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Client
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 10,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Volunteer
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 10,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Guest
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 10,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Professional
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 10,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Grason
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
